tests/test_query.py

```diff
@@ -114,6 +117,72 @@ def mock_httpx_post(**kwargs):
     assert response.status_code == 401
 
 
+def test_app_with_unset_allowed_origins(test_app, monkeypatch):
+    """Tests that when the environment variable for allowed origins has not been set, a warning is raised and the app uses a default value."""
+    monkeypatch.delenv(util.ALLOWED_ORIGINS.name, raising=False)
+    # set random username and password to avoid RuntimeError from other startup check
+    monkeypatch.setenv(util.GRAPH_USERNAME.name, "DBUSER")
+    monkeypatch.setenv(util.GRAPH_PASSWORD.name, "DBPASSWORD")
+
+    with pytest.warns(
+        UserWarning,
+        match=f"API was launched without providing any values for the {util.ALLOWED_ORIGINS.name} environment variable",
+    ):
+        with test_app:
+            pass
+
+    assert util.parse_origins_as_list(
+        os.environ.get(util.ALLOWED_ORIGINS.name, "")
+    ) == [""]
+
+
+@pytest.mark.parametrize(
+    "allowed_origins, parsed_origins, expectation",
+    [
+        (
+            "",
+            [""],
+            pytest.warns(
+                UserWarning,
+                match=f"API was launched without providing any values for the {util.ALLOWED_ORIGINS.name} environment variable",
+            ),
+        ),
+        (
+            "http://localhost:3000",
+            ["http://localhost:3000"],
+            warnings.catch_warnings(),
+        ),
+        (
+            "http://localhost:3000 https://localhost:3000",
+            ["http://localhost:3000", "https://localhost:3000"],
+            warnings.catch_warnings(),
+        ),
+    ],
+)
+def test_app_with_set_allowed_origins(
+    test_app, monkeypatch, allowed_origins, parsed_origins, expectation
+):
+    """
+    Test that when the environment variable for allowed origins has been explicitly set, the app correctly parses it into a list
+    and raises a warning if the value is an empty string.
+    """
+    monkeypatch.setenv(util.ALLOWED_ORIGINS.name, allowed_origins)
+    # set random username and password to avoid RuntimeError from other startup check
+    monkeypatch.setenv(util.GRAPH_USERNAME.name, "DBUSER")
+    monkeypatch.setenv(util.GRAPH_PASSWORD.name, "DBPASSWORD")
+
+    with expectation:
+        with test_app:
```

Comment:
I know this isn't a new pattern but I forgot why we do that. Since I also couldn't find it in our internal wiki, would you mind reminding me why the `with test_app:` bit is needed to catch exceptions and then add it to the wiki, maybe here: https://github.com/neurobagel/documentation/wiki/API ?

Reply:
This pattern is actually documented in FastAPI's own docs, https://fastapi.tiangolo.com/advanced/testing-events/ :)

Reply:
heyho, that's great. Thanks, I missed that